src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/utils/user_config_validatator.py

```diff
@@ -74,6 +101,65 @@ def block_DHCP_config(self):
+            return False
+        if not self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    #checks if only networkForApplianceVM and networkCIDRForApplianceVM are present in config
+    def check_if_nw_config_v2(self):
+        if self.__config["applianceControlPlaneIpAddress"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    def get_config_version(self):
+        if(self.check_if_nw_config_v1()):
+            return Constant.CONFIG_VERSION_V1
+        if(self.check_if_nw_config_v2()):
+            return Constant.CONFIG_VERSION_V2
+        raise InvalidInputError("The Config file does not qualify for either of versions. Please provide the fields according to documentation.")
+    
+    '''
+    Checks if the segment details given by the user in config matches with the segments present in the sddc.
+    '''
+    def validate_segment_details_config(self):
+        res = self._segment_helper.get_segment_list(self.__config['subscriptionId'], 
+                                                    self.__config['resourceGroup'], self.__config['privateCloud'])
+        
+        segment_in_config = self.__config["staticIpNetworkDetails"]["networkForApplianceVM"]
+        segment_cidr_in_config = self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"]
+        for segment in res["value"]:
+            if segment["name"].casefold() == segment_in_config.casefold():
+                if segment["properties"]["subnet"]["gatewayAddress"] != segment_cidr_in_config:
+                    raise InvalidInputError("Segment {} already exists with a different gateway ip cidr".format(segment_in_config))
+
+        for segment in res["value"]:
+            if segment["properties"]["subnet"]["gatewayAddress"] == segment_cidr_in_config:
+                if segment["name"].casefold() != segment_in_config.casefold():
+                    raise InvalidInputError("A different segment already present with gateway ip cidr {}".format(segment_cidr_in_config))
+     
     def validate_avs_config(self):
         self.validate_nw_config()
-        self.validate_azure_details()
\ No newline at end of file
+        self.validate_azure_details()
+        self.validate_segment_details_config()
```

Comment:
If the expected config is changing, we should either updatre the config file or add a new config file ( IMO we should update the existing config file as we want cx to use that)